### ISSUE TITLE
Holo pull rate in pre-2016 core sets are too low

### ISSUE THREAD

**Issue by threwnie** [AlanOC91/YGOPRODeck#294]
I decided to do a comparison of pack odds from the TCG irl and compare them to the pull rates found on this simulator. I tried looking in the repository to see how the sim worked, but due to the closed source nature of the simulator I didn't have an accurate means of comparison besides actually sampling and testing the packs odds manually.

So that's what I did.

**Here's my methodology:**
- Opened 348 packs
- Marked each time I pulled a holo card
- Marked what rarity it was
- Observed the total number of pulls per rarity, & compared them against official TCG pull rates for pre-2016 core sets

For reference, here's the official pack odds for a large majority of pre-2016 TCG boosters (i.e. from Tactical Evolution until Dimension of Chaos):
- Super Rare (1/5; 20%)
- Ultra Rare (1/12; 8.33%)
- Secret Rare (1/23; 4.35%)

For the purposes of my analysis, I excluded Ultimate Rares & Ghost Rares (I've treated them as 'No Holo' in my analysis below) because there's no official pull rates on them. Thus, if we exclude Ultis/Ghosts, the rough odds of pulling a Super/Ultra/Secret holo from any pack should be 32.68%.

**Here's my results from the sim:**
- Packs opened: 348
- Super Rares: 54 (15.52%)
- Ultra Rares: 19 (5.46%)
- Secret Rares: 2 (0.58%)
- Ultimate Rares: 8 (2.30%)
- Ghost Rares: 2 (0.57%)
- Total Holos: 85 (24.43%)

To prove that I wasn't just running on bad luck (although a sample size of 348 should prove otherwise), I also performed a chi square test to prove my results were statistically significant (and because I'm a stats nerd):
![image](https://user-images.githubusercontent.com/76097649/130014713-d6627d0a-95ba-4b03-9b33-5a14b359850a.png)

The p value extracted from this was <0.0001, i.e. **extremely statistically significant**.

All rarities underperformed, though notably problematic were the rate of Secret Rares, which were significantly lower than they should be. My theory is that Ghost Rares and Secret Rares use the same weighting system in the sim, thus leading them to have the same opening rate. Even anecdotally, people who played in that era of Yugioh would often have 1 Secret Rare per 24 pack booster box, rather than 2 Secrets per 348 packs, lol.

I'm not sure why pull rates are lower than what they are IRL, but I hope it's just an error. If there's an alternate reasoning if they're intentionally lower than they should be, please clarify for me. Overall, it also would be nice to have the pack sim be open source, as this would have allowed me to point out the discrepancies between pull rates without having to perform a statistical analysis to prove they're wrong.
________________________________________________________________________________________________________

**TL;DR:** The pack odds are wrong.

**Comment by AlanOC91** [comment]
Hi there,

Firstly thanks for your detailed breakdown and insight. 

I'll start by saying that we purposely **do not do box odds**. I've maintained this from the beginning of the Pack Simulator. It's more variables to the simulator and further complicates things. Because I don't do box odds, I've had to code it in a way that the pull % is based on the data of a single pack rather than a box of 24/36/etc etc. 

Since the simulator is coded this way, it'll never be 100% identical to the real thing but I do strive to make it as close as possible give the above. 

> My theory is that Ghost Rares and Secret Rares use the same weighting system in the sim

I checked back on my code and this is the case for 2 of my algorithms (which it shouldn't be) so I'll definitely adjust that.

Overall, I appreciate this post and it proves to me (in hard stats) that I need to make adjustments, especially with the rate of Secret Rares which I may have overcompensated for.

**Comment by threwnie** [comment]
Hey, thanks for your response and explanation - I appreciate the work you guys put in so glad to help however I can. I'm happy to take another look afterwards if you decide to change up the pull rates.

**Comment by threwnie** [comment]
Hey, just wondering if there's been any update to pull rates - sorry to bump this again.
